packages/xdg_directories/test/xdg_directories_test.dart

```diff
@@ -14,26 +14,26 @@ import 'package:xdg_directories/xdg_directories.dart' as xdg;
 
 void main() {
   final Map<String, String> fakeEnv = <String, String>{};
-  Directory tmpDir;
+  Directory? tmpDir;
```

Comment:
If this is required to be set, as suggested by your widespread use of `!`, then it should be `late`.

Reply:
In the `tearDown` function, we are checking it to be non-null, that's why I didn't use `late` for `tmpDir`.

Reply:
It's never assigned again after `setUp`, so either all the places where you are using `!' are incorrect, because that's a guarantee that you know it's not null, or the `tearDown` check is unnecessary and can be removed. It's impossible for the variable to both be known not to be null and also need to be null checked.